src/SourceLocator/Type/ClosureSourceLocator.php

```diff
@@ -131,17 +133,15 @@ public function leaveNode(Node $node)
              */
             public function getClosureNodes(): array
             {
-                $closureNodesDataOnSameLine = array_values(array_filter($this->closureNodes, fn (array $nodes): bool => $nodes['node']->getLine() === $this->startLine));
-
-                if (! $closureNodesDataOnSameLine) {
+                if ($this->closureNodes === []) {
                     throw NoClosureOnLine::create($this->fileName, $this->startLine);
                 }
 
-                if (isset($closureNodesDataOnSameLine[1])) {
+                if (isset($this->closureNodes[1])) {
```

Comment:
With short closures becoming more popular, I feel like this area will just bite us in future. Refactoring done so far is good as-is though.